hoomd/update/particle_filter.py

```diff
@@ -0,0 +1,92 @@
+"""Implement Python interface for a ParticleFilter updater."""
+
+import copy
+
+import hoomd
+
+
+class _GroupConverter:
+    """Used to go from filter to group for synced list Python <-> C++ syncing.
+
+    We have to use a callable with state because the callable needs to know how
+    to get the ParticleGroup associated with a filter. As the groups are not a
+    user facing object, we need to use the internal group storage of the state
+    object which is available until adding/attaching.
+    """
+
+    def __call__(self, filter):
+        return self._state._groups[type(filter)][filter]
+
+    def _attach(self, simulation):
+        self._state = simulation.state
+
+    def _detach(self):
+        self._state = None
+
+    def __getstate__(self):
+        """Necessary to pickle FilterUpdater."""
+        state = copy.copy(self.__dict__)
+        state.pop("_state", None)
+        return state
+
+
+class FilterUpdater(hoomd.operation.Updater):
+    """Update sets of particles associated with a filter.
+
+    HOOMD stores the particles selected by `hoomd.filter.ParticleFilter`
+    instance that are used by it's objects to avoid recomputing them on demand.
```

Comment:
```suggestion
    HOOMD caches the particles selected by `hoomd.filter.ParticleFilter`
    instances to avoid the cost of re-running the filter on every time step.
```
Reword for clarity.